test/unit/ngsiv2/provisioning/updateProvisionedDevices-test.js

```diff
@@ -0,0 +1,411 @@
+                .matchHeader('fiware-servicepath', '/gardens')
+                .post('/v2/entities/SecondMicroLight/attrs', utils.readExampleFile(
+                    './test/unit/ngsiv2/examples/contextRequests/updateProvisionMinimumDevice.json'))
+                .reply(204);
+
+            async.series([
+                iotAgentLib.clearAll,
+                async.apply(request, provisioning3Options)
+            ], done);
+        });
+
+        it('should not raise any error', function(done) {
+            request(optionsUpdate, function(error, response, body) {
+                should.not.exist(error);
+                response.statusCode.should.equal(204);
+                done();
+            });
+        });
+        it('should provision the attributes appropriately', function(done) {
+            request(optionsUpdate, function(error, response, body) {
+                request(optionsGetDevice, function(error, response, body) {
+                    var parsedBody;
+                    should.not.exist(error);
+                    response.statusCode.should.equal(200);
+
+                    parsedBody = JSON.parse(body);
+
+                    parsedBody.attributes.length.should.equal(1);
+                    parsedBody.attributes[0].name.should.equal('newAttribute');
+                    done();
+                });
+            });
+        });
+        it('should create the initial values for the attributes in the Context Broker', function(done) {
+            request(optionsUpdate, function(error, response, body) {
+                should.not.exist(error);
+                contextBrokerMock.done();
+                done();
+            });
+        });
+    });
+
+    describe('When a device is updated to add static attributes', function() {
+        /* jshint camelcase: false */
+
+        var optionsUpdate = {
+                url: 'http://localhost:' + iotAgentConfig.server.port + '/iot/devices/MicroLight2',
+                method: 'PUT',
+                headers: {
+                    'fiware-service': 'smartGondor',
+                    'fiware-servicepath': '/gardens'
+                },
+                json: utils.readExampleFile('./test/unit/examples/deviceProvisioningRequests/updateDeviceStatic.json')
+            },
+            optionsGetDevice = {
+                url: 'http://localhost:' + iotAgentConfig.server.port + '/iot/devices/MicroLight2',
+                method: 'GET',
+                headers: {
+                    'fiware-service': 'smartGondor',
+                    'fiware-servicepath': '/gardens'
+                }
+            };
+
+        beforeEach(function(done) {
+            nock.cleanAll();
+            contextBrokerMock = nock('http://192.168.1.1:1026')
+                .matchHeader('fiware-service', 'smartGondor')
+                .matchHeader('fiware-servicepath', '/gardens')
+                .post('/v2/entities?options=upsert')
```

Comment:
Mock using post without payload. Comment explaining it needed?

Reply:
Fixed with fbc86fa76f4561b711a63053695efd21941cf4e5